services/src/main/java/edu/unc/lib/dl/cdr/services/processing/IndexingService.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2008 The University of North Carolina at Chapel Hill
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.cdr.services.processing;
+
+import edu.unc.lib.dl.acl.service.AccessControlService;
+import edu.unc.lib.dl.acl.util.AgentPrincipals;
+import edu.unc.lib.dl.acl.util.Permission;
+import edu.unc.lib.dl.data.ingest.solr.exception.IndexingException;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.services.IndexingMessageSender;
+import edu.unc.lib.dl.util.IndexingActionType;
+
+/**
+ * Service that manages reindexing of the repository
+ *
+ * @author harring
+ *
+ */
+public class IndexingService {
+    private AccessControlService aclService;
+    private IndexingMessageSender indexingMessageSender;
+
+    /**
+     * Performs an in-place recursive reindexing of an object and its descendants
+     * or a clean reindexing, as indicated by the request
+     *
+     * @param agent security principals of the agent making request
+     * @param objPid the PID where reindexing should begin in the tree
+     * @param inplace whether in-place reindexing has been requested
+     * @throws IndexingException
+     */
+    public void reindexObjectAndChildren(AgentPrincipals agent, PID objectPid, Boolean inplace) {
```

Comment:
You should be able to safely make `inplace` a `boolean` rather than an object in this case, which will save you the null check below.

Reply:
done